### ISSUE TITLE
[A11y Audit] Design System Team: Accessibility audit of the va-back-to-top component

### ISSUE THREAD

**Issue by jamigibbs** [department-of-veterans-affairs/va.gov-team#60812]
## About your team

- Team name: Design System
-	OCTO-DE product owner: Matthew Dingee
-	Product manager: Carol Wong
-	Designer: Lillian Boot & Barb Denney
-	FE engineer: Jami Gibbs
- Accessibility specialist: N/A
- Content writer:  N/A
-	Product/team Slack channel: #platform-design-system
 
### What does your team need support for? Check all that apply.
- [ ] Launching one or more unauthenticated pages, online forms, apps, or tools on VA.gov
- [ ] Making revisions to one or more existing VA.gov pages
- [ ] Accessibility review of wireframes and/or prototypes
- [ ] Accessibility issue (general request)
- [ ] Accessibility PDF review request
- [ ] Accessibility user research support ([submit a separate, additional request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=coforma-terry%2Cangelafowler82&labels=508%2FAccessibility%2C+sitewide+CAIA%2C+sitewide+accessibility%2C+CAIA-a11y-research&template=caia-a11y-research.yaml&title=%5BCAIA+A11y+Research%5D%3A+Team+name%2C+product+name))
- [x] Something else: Accessibility review of a design system web component
 
### Give a brief description of what your team needs support with:

We would like to request an accessibility audit of our `va-back-to-top` web component. We have a ticket in our backlog that we have prioritized:

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/899

> We need to conduct a full accessibility review of the Back to top component. Months ago Josh Kim @joshkimux and Angela Fowler @AngelaFowler82 identified the Back to top component as not materially honest because it should be a link but instead is styled as a button. There may be an issue with the element living outside the DOM, as well.

> The component can be viewed in Storybook [here](https://design.va.gov/storybook/?path=/docs/components-va-back-to-top--default).
 
 
### Will this new product be released incrementally (for instance 25% of users initially)? 
  - [ ] Yes
  - [x] No
 
**Note:** If you checked yes, we'll reach out to discuss details about the content in the react widget.
  
### When do you expect to launch your product to 100% of users?
  
We would like to complete any work necessary within the current 2 week sprint but this is not a blocker for any other work right now.
 
### Will this work be going through the Collaboration Cycle?
- [ ] Yes 
- [x] No
 
#### If yes, please check all of the phases in the Collaboration Cycle you have completed: 
- [ ] Design Intent
- [ ] Midpoint Review
- [ ] Staging
 
### When does this work need to be completed?
- Estimated launch date:
- Estimated staging review date:
- Content, Accessibility, and IA work needed by: 
  
## Supporting artifacts
*Please provide supporting artifacts as available.*

- Web component preview: https://design.va.gov/storybook/?path=/docs/components-va-back-to-top--default
- Original issue: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/899

#### Do you plan to bring this to an upcoming content office hours session?
- [ ] Yes, benefit content office hours (Thursdays, 3:00 p.m. to 3:45 p.m. ET)
- [ ] Yes, health  content office hours (Thursdays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] Yes, unauth office hours (Mondays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] No, but I'd like to schedule time to talk about this request
- [x] No, let's work asynchronously and meet if needed

 **Note:** If we think this work would benefit from a collaborative session with you, we may ask you to bring it to office hours or set up a separate time to meet.
 
### Accessibility Help in Slack
The [#accessibility-help channel](https://dsva.slack.com/archives/C8E985R32) in the DSVA Slack, is also available to you as a resource. Everyone is welcome to ask questions or get help from our a11y (accessibility) specialists. Tag `@Terry Nichols` to get a11y help asap.

## Next steps

**Once you’ve submitted this ticket, please post a link to this issue in the [#sitewide-content-ia](https://dsva.slack.com/channels/sitewide-content-ia) Slack channel and tag `@Terry Nichols`.**

- If you also need engineering support from the Public Websites team, fill out their [intake request form](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=Public+Websites%2C+Sitewide+content&labels=vsa-public-websites%2C+vsa%2C+vsa-public-websites-intake%2C+sitewide-content%2C+needs-grooming&template=public-websites-intake.md&title=%3CType+of+Request%3E+from+%3CTeam%3E).
  
- If you need a page/URL redirected, a URL changed or a vanity URL set up, please submit a [Redirect, URL change, or vanity URL request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=mnorthuis&labels=ia&template=redirect-request.md&title=Redirect+Request)
 
 - If you are creating an [experimental design,](https://design.va.gov/about/contributing-to-the-design-system/experimental-components-and-patterns#what-is-an-experimental-component-or-pattern?) please [also contact the DST. ](https://design.va.gov/about/contributing-to-the-design-system/suggest-an-addition-or-update)
- If you are requesting [a11y support for user research,](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=coforma-terry%2Cangelafowler82&labels=508%2FAccessibility%2C+sitewide+CAIA%2C+sitewide+accessibility%2C+CAIA-a11y-research&template=caia-a11y-research.yaml&title=%5BCAIA+A11y+Research%5D%3A+Team+name%2C+product+name) please complete this form, next.

**Comment by sara-amanda** [comment]
## CAIA Initial Feedback
- **A button may not be visible to screen mag users** (agrees with feedback from @AngelaFowler82 )*
    - **Tested on 6/27/2023 at 200%, 300% and 400% Zoom:** 
        - It shows as an arrow and just reads as button. (Image pictured below.)
        - When the window returns to 100% the text is visible, and when I re-run Axe, no issues are found.![Screenshot 2023-06-27 at 2.45.56 PM.png](https://images.zenhubusercontent.com/63dbe59975ee0eef93e952b2/3bfc192c-8213-49da-a472-9276e3f65b7f)


- **This button should be a link not a button** (agrees with feedback from @joshkimux ^
- The issue of the button span reading as none,[ appears to be fixed](https://www.va.gov/claim-or-appeal-status/). <button type="button"><i aria-hidden="true" role="img"></i><span>Back to top</span></button>
   
### Alternative to Button, Link Examples: 
**The back to top link is focusable and users can trigger it with the enter key, like any other link. It will take users back to the top of the current page.**
- [Scotland Design System - Back to Top on bottom right
](https://designsystem.gov.scot/components/back-to-top/) 
    - Image pictured below.
        - ![Screenshot 2023-06-27 at 11.51.21 AM.png](https://images.zenhubusercontent.com/63dbe59975ee0eef93e952b2/bda212d5-7aac-466b-896a-5660c2d3fc48)
- [Gov.UK - Design System -Back to Top on bottom left](https://design-system.service.gov.uk/community/accessibility-strategy/)
    - Image pictured below.
        - ![Screenshot 2023-06-27 at 11.52.57 AM.png](https://images.zenhubusercontent.com/63dbe59975ee0eef93e952b2/42a17cea-6024-4489-818c-90f58b7ee54a) 
- [N/N: UX - Back to Top Design Guidelines](https://www.nngroup.com/articles/back-to-top/)

## Additional Feedback
### *Labels

[Referencing ticket #58413 from May 15, 2023 below:
](https://github.com/department-of-veterans-affairs/va.gov-team/issues/58413)

> On a narrow screen, the back to top component displays visually as an arrow presumably pointing to the top of the page. The text label is hidden and there is no accessible label. This causes screen readers to announce the button as "unlabeled 0." It would also make it difficult for users of access technologies such as Dragon Naturally Speaking to use the component as they would not know how to refer to it.

### ^Role = Link

[Referencing the ticket #899 comment from Aug. 10, 2022, below](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/899) :

> joshkimux commented on Aug 10, 2022
> @bkjohnson I can break this down! It's a little bit of an exception, but understandably confusing/awkward.
> 
> The main purpose of the back to top component is to go to a location on the page. Because the user is navigating to a place (scrolling) and nothing is being added (like a modal), removed (like deleting a card from the list loop), or state changed (like going into an edit mode), we can think of this as more of a link than a button.
> 
> You can think of this like the skip link or the on this page link. Both of these are links that send focus to a heading and scroll to a different part of the page.
> 
> And on that note... the sending focus part is a bit tricky though! Ideally, we wouldn't need to send focus there at all, but we need to do that because of some browser/assistive technology combinations. So it's kind of more like a work around 😓

cc: @SarahKay8 @coforma-terry @AngelaFowler82 @joshkimux

**Comment by jamigibbs** [comment]
Hi all, I was wondering if there was any other updates for this ticket? I just want to make sure we weren't waiting on something from myself (or DST in general).

**Comment by sara-amanda** [comment]
> Hi all, I was wondering if there was any other updates for this ticket? I just want to make sure we weren't waiting on something from myself (or DST in general).

Hi, Jami! No other updates to the ticket, unless @SarahKay8 would like to add anything or @AngelaFowler82 (who is back on 7/10) would like to add anything else?

**Comment by jamigibbs** [comment]
Thanks @sara-amanda! I've reviewed your research above but I'm not sure I have clear steps of what to do next (as the developer for this component). It's a little bit difficult for me to interpret exactly what we need to change or update in the component but is this the only issue at this point?

- This button should be a link not a button

**Comment by humancompanion-usds** [comment]
I redesigned the component following the Scots (❤️ SCOTLAND!): https://sketch.com/s/73212626-acc4-47e6-acd8-d4e2f5a98372

Are we okay with the transparency level (90%)?

**Comment by jamigibbs** [comment]
@sara-amanda @SarahKay8  Hi both! Sorry for the delay but we have completed the development work for the va-back-to-top component and it is [available in Storybook](https://design.va.gov/storybook/?path=/docs/components-va-back-to-top--default) if either of you had time to confirm the accessibility updates we've made.

**Comment by SarahKay8** [comment]
Hey @jamigibbs! Will review

**Comment by SarahKay8** [comment]
![Image 7-20-23 at 11.46 AM.jpg](https://images.zenhubusercontent.com/623cd4d6774110eeb19922cb/31f97376-2dee-4a73-bddf-1ca60f7a3f21) Confirmed `back-to-top` updates in storybook. Label is also being properly read out via Windows +  JAWS/ Safari + VO. Thanks for your work on this Jami!
